Application/Analyzing_Application_Issues.md

```diff
@@ -0,0 +1,130 @@
+    3. [Event analysis and visualization with Application Insights](https://docs.microsoft.com/en-us/azure/service-fabric/service-fabric-diagnostics-event-analysis-appinsights)
+    4. [Monitor containers with Azure Monitor logs](https://docs.microsoft.com/en-us/azure/service-fabric/service-fabric-diagnostics-oms-containers)
+1. System Event Log
+    1. Application, System, Security
+    1. Operational cluster events
+        1. [Event aggregation and collection using Windows Azure Diagnostics](https://docs.microsoft.com/en-us/azure/service-fabric/service-fabric-diagnostics-event-aggregation-wad)
+        2. [Service Fabric Linux cluster events in Syslog](https://docs.microsoft.com/en-us/azure/service-fabric/service-fabric-diagnostics-oms-syslog)
+1. Performance Counter (CPU, Memory, Disk IO, Disk capacity, Network)
+    1. [How can I monitor performance counters?](https://docs.microsoft.com/en-us/azure/service-fabric/service-fabric-diagnostics-common-scenarios#how-can-i-monitor-performance-counters)
+1. Service Fabric Explorer 
+    1. Any errors/warnings related to apps, eg. EventStore for cluster and applications
+    2. [Cluster system health reports to troubleshoot](https://docs.microsoft.com/en-us/azure/service-fabric/service-fabric-understand-and-troubleshoot-with-system-health-reports)
+
+### Best practices
+
+Recommendations to help Application owner analyze application issues in distributed environment
+
+1. Log liberally in your applications
+    1. Log success as well as failures
+    1. Failure logs should be verbose and capture as much of the context as possible
+1. Use correlation IDs in logs to allow tracing through the application code to pinpoint failure context
+1. Get familiar with common debugging tools and techniques for distributed environment
+1. Get familiar with [Process Dump Analysis by using WinDbg Preview](https://docs.microsoft.com/en-us/windows-hardware/drivers/debugger/debugging-using-windbg-preview)
+1. Setup alerts on monitoring for hot paths and critical design thresholds
+1. Have defined a process for the escalation path
+1. Know your bottlenecks by load and performance testing every release
+
+### Types of application issues where SF CSS team may be able to provide good insights for helping Application team with investigation 
+
+Service Fabric logs collect information about service fabric cluster and very minimal information about the applications themselves. Service Fabric does not have insights into the internals of the application and hence does not have the ability to capture information about the application behavior. Service Fabric has knowledge about the following and can provide some insights to help with application but nor necessarily the exact cause.
+
+- Service lifecycle events i.e. startup, role change, shutdown and any abnormal behavior during these phases
```

Comment:
>- Service lifecycle [](http://example.com/codeflow?start=0&length=19)

other bullet points:
- interaction between the application and the SF runtime (if it exists)
- set up of application resources by the SF runtime (accounts, certificates, endpoints)
- connection handshakes for apps built on the SF SDK

Please note that, with a few exceptions, SF logs are diagnostic traces, and do not constitute an audit.

Reply:
Excellent! Especially the last statement. Thank you.